pennylane/ops/qubit.py

```diff
@@ -43,26 +44,27 @@ class Hadamard(Observable, Operation):
     num_params = 0
     num_wires = 1
     par_domain = None
-    eigvals = pauli_eigs(1)
+    eigvals = lambda self: pauli_eigs(1)
```

Comment:
With this implementation, this seems to be working as a static method as well: calling ``qml.Hadamard.eigvals('a')`` returns the eigenvalues. Was that intentional and it is fine like that or would there be some reason restricting it?

Reply:
I've played around with it, and this seems to just be a side effect of Python's object oriented design. When you create an instance method that does not depend on `self`, Python will not complain when you attempt to call it as a static method with arbitrary arguments passed.

Since we want consistency, I think it is safe to ignore this behaviour.

AFAICT there is no way around this --- gates with no parameters will have eigval/matrix methods that do not depend on `self` (since the eigvals/matrices are known constants), whereas parametrized gates eigvals/matrices _will_ depend on `self`. I also hesitate to separate the structure of the operations into parametrized gate classes and unparametrized gate classes.

Reply:
Currently, I also do not see any reasons why this behavior could be disadvantageous.

The use of this method as a class classmethod, however, would be better avoided I believe. As the parameter could be *any object*, using it could lead to confusing code.

What do you have in mind by "separate the structure of the operations into parametrized gate classes and unparametrized gate classes", having different design decisions for these two types of classes?

Reply:
I think this is now no longer the case, since `eigvals` is a property :slightly_smiling_face: I think this is a good solution to the ambiguity